### PR TITLE
002 Package Strucure and Code Organization

### PR DESCRIPTION
**Pull request by manoranjith**
Proposal for package structure and code organization. Relates to direct-state-transfer/dst-go#38

### CODE REVIEW

**Comment by choeppler** [comment]
@manoranjith, I guess 53a099f slipped into this PR by mistake, didn't it?

**Comment by manoranjith** [comment]
Yes, commit 53a099f slipped in by mistake. I have removed it now.

File: design/002-Package-Structure-Code-Organiation.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+<!-- This is a template for proposing design changes to the dst-go project. -->
+
+# Proposal: <!-- Package structure and code organization for dst-go -->
```
Comment (choeppler) [inline design/002-Package-Structure-Code-Organiation.md:3]: The proposal title should not be commented out.
```suggestion
# Proposal: Package structure and code organization for dst-go
```

Comment (manoranjith) [reply design/002-Package-Structure-Code-Organiation.md]: Updated the changes in this commit - fafe44f.

**Comment by manoranjith** [comment]
Hi @choeppler, 
I have squashed the commits and updated the pull request.
I have also changed the status to `accept` and added source file for the diagram.

Can you please review it again.

**Pull request merged**